physics/resonance_test.cpp

```diff
@@ -0,0 +1,391 @@
+    elements_[laythe_].conic.eccentricity = 0;
+    elements_[laythe_].conic.semimajor_axis = 27'184'000 * Metre;
+    elements_[laythe_].inclination = 0 * Degree;
+    elements_[laythe_].longitude_of_ascending_node = 0 * Degree;
+    elements_[laythe_].argument_of_periapsis = 0 * Degree;
+    elements_[laythe_].mean_anomaly = 3.14 * Radian;
+    elements_[vall_].conic.eccentricity = 0;
+    elements_[vall_].conic.semimajor_axis = 43'152'000 * Metre;
+    elements_[vall_].inclination = 0 * Degree;
+    elements_[vall_].longitude_of_ascending_node = 0 * Degree;
+    elements_[vall_].argument_of_periapsis = 0 * Degree;
+    elements_[vall_].mean_anomaly = 0.9 * Radian;
+    elements_[tylo_].conic.eccentricity = 0;
+    elements_[tylo_].conic.semimajor_axis = 68'500'000 * Metre;
+    elements_[tylo_].inclination = 0.025 * Degree;
+    elements_[tylo_].longitude_of_ascending_node = 0 * Degree;
+    elements_[tylo_].argument_of_periapsis = 0 * Degree;
+    elements_[tylo_].mean_anomaly = 3.14 * Radian;
+    elements_[bop_].conic.eccentricity = 0.24;
+    elements_[bop_].conic.semimajor_axis = 128'500'000 * Metre;
+    elements_[bop_].inclination = 15 * Degree;
+    elements_[bop_].longitude_of_ascending_node = 10 * Degree;
+    elements_[bop_].argument_of_periapsis = 25 * Degree;
+    elements_[bop_].mean_anomaly = 0.9 * Radian;
+    elements_[pol_].conic.eccentricity = 0.17;
+    elements_[pol_].conic.semimajor_axis = 179'890'000 * Metre;
+    elements_[pol_].inclination = 4.25 * Degree;
+    elements_[pol_].longitude_of_ascending_node = 2 * Degree;
+    elements_[pol_].argument_of_periapsis = 15 * Degree;
+    elements_[pol_].mean_anomaly = 0.9 * Radian;
+    parents_.emplace(jool_, sun_);
+    for (auto const moon : joolian_moons_) {
+      parents_.emplace(moon, jool_);
+    }
+    google::LogToStderr();
+  }
+
+  void FixVallMeanAnomaly() {
+    // NOTE(egg): In the stock game, this is 0.9 rad.  This is very close to an
+    // unstable resonance, and as such no interpretation of the orbital elements
+    // will make the system stable for any reasonable length of time.  We make
+    // this a stable resonance instead.
+    elements_[vall_].mean_anomaly = 0 * Radian;
+  }
+
+  // Fills |stock_orbits_|.
+  // KSP assumes secondaries are massless.
+  void ComputeStockOrbits() {
+    for (auto const body : jool_system_) {
+      stock_orbits_.emplace(
+          body,
+          KeplerOrbit<KSP>(*parents_[body],
+                           test_particle_,
+                           game_epoch_,
+                           elements_[body]));
+    }
+  }
+
+  // Keep stock's mean motion when changing the gravitational parameter (instead
+  // of keeping the semimajor axis).
+  void UseStockMeanMotions() {
+    for (auto const body : jool_system_) {
+      elements_[body].conic.semimajor_axis = std::experimental::nullopt;
+      elements_[body].conic.mean_motion = stock_orbits_.at(body).mean_motion();
```

Comment:
I'd rather not use `at`.  Let's stick to [], it does a check in debug mode anyway.